sig-architecture/production-readiness.md

```diff
@@ -28,6 +27,30 @@ aspects of the process.
     happens if it is subsequently upgraded again?
   - Are there tests for this?
 * Scalability
+  - Will enabling / using the feature result in any new API calls?
+    Describe them with their impact keeping in mind the [supported limits][]
+    (e.g. 5000 nodes per cluster, 100 pods/s churn) focusing mostly on:
+     - components listing and/or watching resources they didn't before
+     - API calls that may be triggered by changes of some Kubernetes
+       resources (e.g. update object X based on changes of object Y)
+     - periodic API calls to reconcile state (e.g. periodic fetching state,
+       heartbeats, leader election, etc.)
+  - Will enabling / using the feature result in supporting new API types?
+    How many objects of that type will be supported (and how that translates
+    to limitations for users)?
+  - Will enabling / using the feature result in increasing size or count
+    of the existing API objects?
+  - Will enabling / using the feature result in increasing time taken
```

Comment:
I'm generally concerned about being overly restrictive on this.  Some features are worth a cost and as long as the cost is optional (either out or in), I think that should be willing to describe that cost and the tradeoff to made.  This comes up in cases of admission plugins and features like server-side-apply.  As long as the additional cost is explained, the tradeoff made clear, and the adoption time reasonable, I don't think we should block features that add cost from being promoted to GA.

Reply:
The goal is not to be to restrictive (answer yes is fine, if this is justified). The purpose of these questions is to understand the impact not to block the feature (at least not in all cases).